BaristaSwift/Common/Common/Store/FavoritesStore.swift

```diff
@@ -47,11 +48,17 @@ public class FavoritesStore: Store<UserFavorite> {
         return UserFavorite.from(results)
     }
     
-    public func addNewFavorite(_ forProduct:Product, completion:SyncCompletion) {
-        guard let account = AccountStore.instance.myAccount(), let accountId = account.accountId else { return }
+    public func addNewFavorite(_ forProduct:Product) -> Promise<UserFavorite> {
+        guard let account = AccountStore.instance.myAccount(), let accountId = account.accountId else {
```

Comment:
Again, notice the error handling, if a guard fails, I fail the promise, so the code in the then blocks only deals with the normal case.